static/js/drafts.js

```diff
@@ -365,12 +366,35 @@ function row_with_focus() {
 
 function row_before_focus() {
     const $focused_row = row_with_focus();
-    return $focused_row.prev(".draft-row:visible");
+    const $prev_row = $focused_row.prev(".draft-row:visible");
+    // The draft modal can have two sub-sections. This handles the edge case
+    // when the user moves from the second "Other drafts" section to the first
+    // section which contains drafts from a particular narrow.
+    if (
+        $prev_row.length === 0 &&
+        $focused_row.parent().attr("id") === "other-drafts" &&
+        $("#drafts-from-conversation").is(":visible")
+    ) {
+        return $($("#drafts-from-conversation").children(".draft-row:visible").last());
+    }
+
+    return $prev_row;
 }
 
 function row_after_focus() {
     const $focused_row = row_with_focus();
-    return $focused_row.next(".draft-row:visible");
+    const $next_row = $focused_row.next(".draft-row:visible");
+    // The draft modal can have two sub-sections. This handles the edge case
+    // when the user moves from the first section (drafts from a particular
+    // narrow) to the second section which contains the rest of the drafts.
+    if (
+        $next_row.length === 0 &&
+        $focused_row.parent().attr("id") === "drafts-from-conversation" &&
+        $("#other-drafts").is(":visible")
+    ) {
+        return $($("#other-drafts").children(".draft-row:visible")[0]);
```

Comment:
Should this use `.first()` for parity with `.last()` above? Seems more readable in any case.

Reply:
It can't sadly, because the header is the first child. I can add a comment about that if that's helpful.

Reply:
Yeah, a comment would be a good choice here. Though ... I'd have expected the header not to have the `draft-row` CSS class.

Reply:
Ohh oops you're right. I think what happened is I used `.last()` without filtering children, realized I needed to filter in this function, and then went back and added the filter before `.last()` so that they'd match up... and then missed the final step of switching this to `first()` -- thanks for your patience!

Reply:
Yeah, that makes sense!